kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -393,63 +407,45 @@ protected void handleInactive(KafkaHeaderAndRequest kafkaHeaderAndRequest,
         return admin.topics().getPartitionedTopicMetadataAsync(topicName);
     }
 
-    // Get all topics exclude `__consumer_offsets` and `__transaction_state`, the result of `topicMapFuture` is a map:
+    private boolean isInternalTopic(final String fullTopicName) {
+        return fullTopicName.equals(offsetsTopicName) || fullTopicName.equals(txnTopicName);
+    }
+
+    // Get all topics in the configured allowed namespaces.
     //   key: the full topic name without partition suffix, e.g. persistent://public/default/my-topic
     //   value: the partitions associated with the key, e.g. for a topic with 3 partitions,
     //     persistent://public/default/my-topic-partition-0
     //     persistent://public/default/my-topic-partition-1
     //     persistent://public/default/my-topic-partition-2
     private void getAllTopicsAsync(CompletableFuture<Map<String, List<TopicName>>> topicMapFuture) {
-        final String offsetsTopicName = new KopTopic(String.join("/",
-                kafkaConfig.getKafkaMetadataTenant(),
-                kafkaConfig.getKafkaMetadataNamespace(),
-                GROUP_METADATA_TOPIC_NAME)
-        ).getFullName();
-        final String txnTopicName = new KopTopic(String.join("/",
-                kafkaConfig.getKafkaMetadataTenant(),
-                kafkaConfig.getKafkaMetadataNamespace(),
-                TRANSACTION_STATE_TOPIC_NAME)
-        ).getFullName();
-        admin.tenants().getTenantsAsync().thenApply(tenants -> {
-            if (tenants.isEmpty()) {
-                topicMapFuture.complete(Maps.newHashMap());
-                return null;
-            }
-            Map<String, List<TopicName>> topicMap = Maps.newConcurrentMap();
+        final Map<String, List<TopicName>> topicMap = new ConcurrentHashMap<>();
+        final AtomicInteger pendingNamespacesCount = new AtomicInteger(allowedNamespaces.size());
+
+        for (String namespace : allowedNamespaces) {
+            pulsarService.getNamespaceService().getListOfPersistentTopics(NamespaceName.get(namespace))
+                    .whenComplete((topics, e) -> {
+                        if (e != null) {
```

Comment:
We might get namespaces not exist here? If yes, we should not failed the request here.

Reply:
If the namespace doesn't exist, in the callback, `topics` will be empty and `e` is still null. I'll add the tests.

Reply:
I mean one namespace does not exist because here is check all the namespaces. Such as we have 10 namespaces but when list topics for a namespace of the 10 namespaces, we got a namespace not exist exception here, we should return the topic list of the other 9 namespaces.